File: src/leetcode/2026/04/Solution824.cpp
#include <string>
#include <unordered_set>
#include <sstream>
#include <iostream>

using namespace std;

class Solution824 {
public:
    string toGoatLatin(string sentence) {
        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        stringstream ss(sentence);
        string word;
        string result;
        int index = 1;
        
        while (ss >> word) {
            if (vowels.find(word[0]) != vowels.end()) {
                word += "ma";
            } else {
                word = word.substr(1) + word[0] + "ma";
            }
            word += string(index, 'a');
            if (index > 1) result += " ";
            result += word;
            index++;
        }
        
        return result;
    }
};

int main() {
    Solution824 solution;

    string sentence = "I speak Goat Latin";
    string result = solution.toGoatLatin(sentence);
    // Expected output: "Imaa peaksmaaa oatGmaaaa atinLmaaaaa"
    cout << result << endl;

    sentence = "The quick brown fox jumped over the lazy dog";
    result = solution.toGoatLatin(sentence);
    // Expected output: "heTmaa uickqmaaa rownbmaaaa oxfmaaaaa umpedjmaaaaaa overmaaaaaaa hetmaaaaaaaa azylmaaaaaaaaa ogdmaaaaaaaaaa"
    cout << result << endl;

    return 0;
}

File: src/leetcode/2026/04/Solution824.java
public class Solution824 {
    public String toGoatLatin(String sentence) {
        String[] words = sentence.split(" ");
        StringBuilder result = new StringBuilder();
        String vowels = "aeiouAEIOU";
        for (int i = 0; i < words.length; i++) {
            String word = words[i];
            if (vowels.indexOf(word.charAt(0)) != -1) {
                result.append(word);
            } else {
                result.append(word.substring(1)).append(word.charAt(0));
            }
            result.append("ma");
            for (int j = 0; j <= i; j++) {
                result.append("a");
            }
            if (i < words.length - 1) {
                result.append(" ");
            }
        }
        return result.toString();
    }

    public static void main(String[] args) {
        Solution824 solution = new Solution824();

        String sentence1 = "I speak Goat Latin";
        String result1 = solution.toGoatLatin(sentence1);
        System.out.println(result1); // Output: "Imaa peaksmaaa oatGmaaaa atinLmaaaaa"

        String sentence2 = "The quick brown fox jumped over the lazy dog";
        String result2 = solution.toGoatLatin(sentence2);
        System.out.println(result2); // Output: "heTmaa uickqmaaa rownbmaaaa oxfmaaaaa umpedjmaaaaaa overmaaaaaaa
                                     // hetmaaaaaaaa azylmaaaaaaaaa ogdmaaaaaaaaaa"
    }
}
